src/HardwareNFCConnectionPN532.cpp

```
@@ -143,21 +143,32 @@ uint16_t HardwareNFCConnectionPN532::readISO14443Pages(
+    // The NTAG READ command behind mifareultralight_ReadPage always returns
+    // 16 bytes (4 pages); the library keeps 4 and leaves the whole response in
+    // its file-scope packet buffer (frame: [7]=status, [8..23]=data), which
+    // this file already scrapes for ATQA/SAK. Harvesting all 16 bytes per
+    // round-trip cuts radio exchanges 4x — a 10-page classify read costs 3
+    // exchanges instead of 10, a 50-page NDEF read 13 instead of 50 (#242).
     uint16_t bytesRead = 0;
-    for (uint8_t i = 0; i < pageCount; i++) {
-        uint8_t page = startPage + i;
+    for (uint16_t i = 0; i < pageCount; i += 4) {
+        uint8_t page = startPage + (uint8_t)i;
         uint8_t pageBuf[4];
```

Clang (14.0.6)

[warning] do not declare C-style arrays, use std::array<> instead

(modernize-avoid-c-arrays)